### DISCUSSION TITLE
sry but i cannot find dart type in rust

### DISCUSSION THREAD

**Discussion by HeavySnowJakarta** [fzyzcjy/flutter_rust_bridge#1887] [Q&A]
maybe one of the most basic problems. i defined a type with dart to do the file io operations:

```
abstract class FileIOManagerDart{
  late Directory workDict;
  late String workPath;
  Future<String> read(String path);
  Future<void> write(String path, String content);
}
```

i really have no idea to use it in another `rs` file. should i `use` something? to use it directly in the `rs` file the lsp server just tells `cannot find type \`FileIOManagerDart\` in this scope`. how to call the dart methods within the rust functions?

**Comment by fzyzcjy** [comment]
Hi, it is not implemented (yet), but as a workaround, you can pass any Dart functions (closures) to Rust side https://cjycode.com/flutter_rust_bridge/guides/direction/rust-call-dart
